gateway-modules/apim-ballerina-configs/configs/org/wso2/carbon/apimgt/gateway/apimPublisherEventListner.bal

```diff
@@ -74,8 +74,16 @@ service jmsService {
                     system:println("Invalid json received");
                 }
 
-            } else if(strings:equalsIgnoreCase(eventType, Constants:API_PUBLISH)){
-                //TODO: Implement logic
+            } else if(strings:equalsIgnoreCase(eventType, Constants:API_STATE_CHANGE)){
+                json apiSummary = jsons:getJson(event, "apiSummary");
+                if(apiSummary != null){
+
+                //Update API cache
+                dto:APIDTO api = gatewayUtil:fromJSONToAPIDTO(apiSummary);
+                holder:putIntoAPICache(api);
+                } else {
+                system:println("Invalid json received");
```

Comment:
Fixed in https://github.com/wso2/carbon-apimgt/pull/4087/commits/e9aae8173ce0104f16c45795dee53026a884e801